### ISSUE TITLE
Publish coreutils/uutils to crates.io

### ISSUE THREAD

**Issue by PlatonB** [uutils/coreutils#1337]
OS: Fedora 28.

```
cargo install uutils
    Updating crates.io index
error: could not find `uutils` in registry `https://github.com/rust-lang/crates.io-index`
```

```
cargo install coreutils
    Updating crates.io index
error: could not find `coreutils` in registry `https://github.com/rust-lang/crates.io-index`
```

**Comment by Arcterus** [comment]
That's because `uutils` is not on crates.io at the moment.  We'll hopefully have a version up relatively soon.

**Comment by sylvestre** [comment]
@Arcterus 
any chance you could make this happen ? I would like to package coreutils for Debian&Ubuntu but I need the crates to be crates.io
thanks

**Comment by sylvestre** [comment]
First, good news, I have been able to get access to https://crates.io/crates/coreutils

Then, I had a look to the current crate to see what would it take to publish it.

Currently, if we want to publish uutils, we would have to publish all internal dependencies (arch, ls, cat, etc) to crates.io. I don't think we want this (and some names are already used by others). 
As @SimonSapin told me, a Cargo.toml = one page crates.io = one cargo publish

This is done by other projects: https://github.com/gluon-lang/gluon/blob/5b81e0c2eff5037412788c0b4ce8fbbf62a86d4b/scripts/publish.sh 

Example here:
```
% cargo publish --dry-run
    Updating crates.io index
   Packaging uutils v0.0.1 (/home/sylvestre/dev/debian/coreutils)
error: failed to prepare local package for uploading

Caused by:
  failed to select a version for the requirement `base32 = "^0.0.1"`
  candidate versions found which didn't match: 0.4.0, 0.3.1, 0.3.0, ...
  location searched: crates.io index
required by package `uutils v0.0.1 (/home/sylvestre/dev/debian/coreutils)`
```

One other solution might be to rethink the whole build infra of the crate to have one Cargo to build everything at once (each binary)... maybe even stop uutils as a binary.

**Comment by rivy** [comment]
Well, that's ... awkward and painful.

None of the internal crates are self-contained, so, even publishing them would take some significant refactoring.

And, on top of that, there aren't any crates.io namespaces, so we'd never be able to group them without someone name-squatting right in the middle of the grouping.

Alternatively, converting everything to modules ... I'm not really sure what that would look like ... Yikes!

@uutils/members , Any ideas?

**Comment by rivy** [comment]
Fixed by 1c88bc8c.
